File: 1364-tuple-with-same-product/tuple-with-same-product.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& nums) {
        int n=nums.size();
        unordered_map<int,int>mp;
        int ans=0;
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                int sm=nums[i]*nums[j];
               // cout<<sm<<" ";
                
                mp[sm]++;
            }
           
        }
        for(auto it:mp){
            if(it.second>1){
                int tmp=(it.second*(it.second-1))/2;
                ans+=(tmp*8);
            }
        }
        return ans;

    }
};

File: 1364-tuple-with-same-product/tuple-with-same-product.py
class Solution(object):
    def tupleSameProduct(self, nums):
        n=len(nums)
        mp={}
        
        ans=0
        for i in range(n):
            for j in range(i+1,n):
                sm=nums[i]*nums[j]
                mp[sm]=mp.get(sm,0)+1


        for x in mp:
            if mp[x]>1:
                tmp=(mp[x]*(mp[x]-1))/2
                ans+=(tmp*8)

        return ans

        